sm/tedge_agent/src/operation_logs.rs

```diff
@@ -0,0 +1,199 @@
+use chrono::{SecondsFormat, Utc};
+use plugin_sm::log_file::LogFile;
+use std::cmp::Reverse;
+use std::collections::BinaryHeap;
+use std::path::PathBuf;
+use tracing::log;
+
+#[derive(Debug)]
+pub struct OperationLogs {
+    log_dir: PathBuf,
+    max_update_logs: usize,
+    max_list_logs: usize,
+}
+
+pub enum LogKind {
+    SoftwareUpdate,
+    SoftwareList,
+}
+
+const UPDATE_PREFIX: &'static str = "software-update";
+const LIST_PREFIX: &'static str = "software-list";
+
+impl OperationLogs {
+    pub fn try_new(log_dir: PathBuf) -> Result<OperationLogs, std::io::Error> {
+        std::fs::create_dir_all(log_dir.clone())?;
+        let operation_logs = OperationLogs {
+            log_dir,
+            max_update_logs: 5,
+            max_list_logs: 1,
+        };
+
+        if let Err(err) = operation_logs.remove_outdated_logs() {
+            log::warn!("Fail to remove the out-dated log files: {}", err);
+        }
```

Comment:
Is the constructor the right place to call this method? Maybe this logic should be delegated to the struct user, currently it is used in Agent constructor so failure to remove old logs will prevent agent running.

Reply:
I prefer to have this cleaning logic implicit.

You are correct, in no case a log-cleaning error should prevent the agent to run. This is why the error is logged but not returned.